#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.06660386160289378 0.9628453831132167 0.2617110884871825 2.660985987319877
  position 1.9888946604294437 3.032917250617149 -3.801204083474306
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
DEF OMNI_WHEELS Robot {
  translation 0 0.06 0
  children [
    DEF SOLID1 Solid {
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "wheel1"
              maxVelocity 52.3598
              maxTorque 0.5931
            }
            PositionSensor {
              name "encoder1"
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.02
                            radius 0.01
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.678431 0.847059 0.901961
                      roughness 1
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.05
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 1"
      boundingObject DEF DUMMY Sphere {
        radius 0.01
      }
      physics Physics {
      }
    }
    DEF SOLID2 Solid {
      rotation 0 1 0 2.09
      children [
        DEF WHEEL2 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "wheel2"
              maxVelocity 52.4
              maxTorque 0.5931
            }
            PositionSensor {
              name "encoder2"
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0.00240580999995935
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 2"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF SOLID3 Solid {
      rotation 0 1 0 4.18
      children [
        DEF WHEEL3 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "wheel3"
              maxVelocity 52.4
              maxTorque 0.5931
            }
            PositionSensor {
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0.0039094400000208316
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 3"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF BODY_1 Group {
      children [
        DEF BODY_TRANS Transform {
          translation 0 0.03 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.564706 0.933333 0.564706
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.15
                radius 0.1
                subdivision 18
              }
            }
          ]
        }
        DEF D_1 DistanceSensor {
          translation 0.11662417886764999 0.07999999999999999 0.014311550458618361
          rotation 0 1 0 -0.4363
          children [
            DEF S_SENSOR Shape {
              appearance PBRAppearance {
                baseColor 0.4470588235294118 0.6235294117647059 0.8117647058823529
              }
              geometry Box {
                size 0.015 0.02 0.043
              }
            }
          ]
          name "DSENSOR_F1"
          lookupTable [
            0 0 0
            0.02 0 0
            5 1660 0
          ]
          resolution 1
        }
        DEF D_2 DistanceSensor {
          translation 0.06926644471302822 0.08 0.09585407141817587
          rotation 0 1 0 -0.6107
          children [
            DEF S_SENSOR Shape {
              appearance PBRAppearance {
                baseColor 0.4470588235294118 0.6235294117647059 0.8117647058823529
              }
              geometry Box {
                size 0.015 0.02 0.043
              }
            }
          ]
          name "DSENSOR_F2"
          lookupTable [
            0 0 0
            0.02 0 0
            5 1660 0
          ]
          resolution 1
        }
      ]
    }
  ]
  boundingObject USE BODY_TRANS
  physics Physics {
  }
  controller "U3_Exam_May_Alexis"
}
Solid {
  translation 2 0.25 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry DEF G_M Box {
        size 0.05 0.5 4
      }
    }
    Solid {
      translation -4 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry USE G_M
        }
      ]
      boundingObject USE G_M
      physics Physics {
      }
    }
    Solid {
      translation -2 0 2
      rotation 0 1 0 1.571
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry USE G_M
        }
      ]
      name "solid(1)"
      boundingObject USE G_M
      physics Physics {
      }
    }
    Solid {
      translation -2 0 -2
      rotation 0 1 0 1.571
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry USE G_M
        }
      ]
      name "solid(2)"
      boundingObject USE G_M
      physics Physics {
      }
    }
  ]
  boundingObject USE G_M
  physics Physics {
  }
}
